osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+from datetime import date
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+
+
+def deleteScansResults(scansToDelete):
+    from kobo.hub import models
+    for scan in scansToDelete:
+        try:
+            pathToDelete = models.Task.get_task_dir(scan.id)
+            if os.path.isdir(pathToDelete):
+                shutil.rmtree(pathToDelete)
+        except OSError as e:
+            logging.error(f"Error deleting scans: {e}")
+
+
+# In order to identify personal scans, we are using the owner field. If the owner doesn't contain any / or @, we assume
+# that is a personal scan
+def identifyPersonalScan(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough and we are more perfomance-oriented.
+    if "/" in owner or "@" in owner:
+        return False
+    else:
+        return True
+
+
+def main():
+    import django
+    django.setup()
+
+    # Retention policy Scans
+    print("Starting Retention Policy Enforcement")
+
+    from osh.hub.scan import models
+    tasks = models.Task.objects.all()
+
+    # ---------- Section to take parameters from DB
+
+    retentionPersonalScans = 14
+    retentionFailedStatus = 14
+
+    if models.RetentionPolicySetting.objects.filter(category="PersonalScan").exists():
+        retentionPersonalObject = models.RetentionPolicySetting.objects.get(category="PersonalScan")
+        retentionPersonalScans = retentionPersonalObject.days
+    else:
+        # We set up the default time
+        retentionPersonalScans = 14
+        logging.warning("Error getting PersonalScan parameter. Setting default parameter")
+
+    if models.RetentionPolicySetting.objects.filter(category="FailedStatus").exists():
+        retentionFailedObject = models.RetentionPolicySetting.objects.get(category="FailedStatus")
+        retentionFailedStatus = retentionFailedObject.days
+    else:
+        # We set up the default time
+        retentionFailedStatus = 14
```

Comment:
This variable was already set to the same value on line 52. Thus, this assignment is redundant.